kubectl-openebs/cli/command/describe/volume_info.go

```diff
@@ -99,112 +98,109 @@ func RunVolumeInfo(cmd *cobra.Command, vols []string, ns string) error {
-		Namespace:               volumeInfo.Namespace,
-		PVC:                     pvInfo.Spec.ClaimRef.Name,
-		ReplicaCount:            volumeInfo.Spec.ReplicationFactor,
-		VolumePhase:             pvInfo.Status.Phase,
-		StorageClass:            pvInfo.Spec.StorageClassName,
-		Version:                 util.CheckVersion(volumeInfo.VersionDetails),
-		Size:                    volumeInfo.Status.Capacity.String(),
-		Status:                  volumeInfo.Status.Phase,
-	}
-
-	// Print the output for the portal status info
-	tmpl, err := template.New("volume").Parse(volInfoTemplate)
-	if err != nil {
-		return errors.Wrap(err, "error displaying output for volume info")
-	}
-	err = tmpl.Execute(os.Stdout, volume)
-	if err != nil {
-		return errors.Wrap(err, "error displaying volume details")
-
-	}
-
-	portalInfo := util.PortalInfo{
-		IQN:            volumeInfo.Spec.Iqn,
-		VolumeName:     volumeInfo.Name,
-		Portal:         volumeInfo.Spec.TargetPortal,
-		TargetIP:       volumeInfo.Spec.TargetIP,
-		TargetNodeName: NodeName,
-	}
-
-	// Print the output for the portal status info
-	tmpl, err = template.New("PortalInfo").Parse(portalTemplate)
-	if err != nil {
-		return errors.Wrap(err, "error creating output for portal info")
-	}
-	err = tmpl.Execute(os.Stdout, portalInfo)
-	if err != nil {
-		fmt.Println(err, "error displaying target portal detail")
-		return nil
-	}
-
-	replicaCount := volumeInfo.Spec.ReplicationFactor
-	// This case will occur only if user has manually specified zero replica.
-	// or if none of the replicas are healthy & running
-	if replicaCount == 0 || len(volumeInfo.Status.ReplicaStatuses) == 0 {
-		fmt.Println("None of the replicas are running")
-		//please check the volume pod's status by running [kubectl describe pvc -l=openebs/replica --all-namespaces]\Oor try again later.")
-		return nil
-	}
-
-	// Print replica details
-	fmt.Printf("Replica Details :\n----------------\n")
-	out := make([]string, len(cvrInfo.Items)+2)
-	out[0] = "Name|Pool Instance|Status"
-	out[1] = "----|-------------|------"
-	for i, cvr := range cvrInfo.Items {
-		out[i+2] = fmt.Sprintf("%s|%s|%s",
-			cvr.ObjectMeta.Name,
-			cvr.Labels[cstortypes.CStorPoolInstanceNameLabelKey],
-			cvr.Status.Phase,
-		)
+	for _, volName := range vols {
+		// Fetch all details of a volume is called to get the volume controller's
+		// info such as controller's IP, status, iqn, replica IPs etc.
+		//1. cStor volume info
+		volumeInfo, err := clientset.GetcStorVolume(volName)
+		if err != nil {
+			fmt.Printf("failed to get CStorVolume %s\n", volName)
+		}
+		//2. Persistent Volume info
+		pvInfo, err := clientset.GetPV(volName)
+		if err != nil {
+			fmt.Printf("failed to get PV %s\n", volName)
+		}
+		//3. cStor Volume Config
+		cvcInfo, err := clientset.GetCVC(volName)
+		if err != nil {
+			fmt.Printf("failed to get cStor Volume config for %s", volName)
+		}
+
+		//4. Get Node for Target Pod from the openebs-ns
+		nodeName, err := clientset.NodeForVolume(volName)
```

Comment:
- This logic would show the node name of the target pod which is not important for the user as it is an internal resource.
- We should display the `cva` ownerNodeID here, as that is relevant to the application using the volume.

Reply:
👍🏼 I'll set it to `N/A` in case the CVA isn't found for this resource.